### PR TITLE
 Add scale-up test from 0 for GPU node-pool

### PR DESCRIPTION
**Pull request by losipiuk**
**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 1]
```diff
@@ -207,6 +207,38 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
...
+		framework.SkipUnlessProviderIs("gke")
+
+		const gpuPoolName = "gpu-pool"
+		addGpuNodePool(gpuPoolName, "nvidia-tesla-k80", 1, 0)
+		defer deleteNodePool(gpuPoolName)
+		framework.ExpectNoError(enableAutoscaler(gpuPoolName, 0, 1))
+		Expect(len(getPoolNodes(f, gpuPoolName)), 0)
+
+		// add deamon set to register drivers
+		// TODO: why gke is not doing this? Which version should be use
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: Can you rephrase this into actionable TODO? Something like:
```TODO(<your username>): <what needs to be than and when>```

As for which version, my guess would be matching K8s version, but we should probably check this with accelerators team

File: test/utils/runners.go [hunk 2]
```diff
@@ -633,6 +634,10 @@ func (config *RCConfig) applyTo(template *v1.PodTemplateSpec) {
 	if config.MemRequest > 0 {
 		template.Spec.Containers[0].Resources.Requests[v1.ResourceMemory] = *resource.NewQuantity(config.MemRequest, resource.DecimalSI)
 	}
+	if config.GpuRequest > 0 {
+		template.Spec.Containers[0].Resources.Limits["nvidia.com/gpu"] = *resource.NewQuantity(config.GpuRequest, resource.DecimalSI)
+		// TODO[losipiuk] should I use framework.NVIDIAGPUResourceName from gpu_util.go as key?
```
Comment (aleksandra-malinowska) [inline test/utils/runners.go]: If we have this constant defined somewhere in the codebase, we should probably use it. 

Can you either remove this TODO, or rephrase as suggested above?

File: test/utils/runners.go [hunk 3]
```diff
@@ -124,6 +124,7 @@ type RCConfig struct {
 	CpuLimit          int64 // millicores
 	MemRequest        int64 // bytes
 	MemLimit          int64 // bytes
+	GpuRequest        int64 // count
```
Comment (aleksandra-malinowska) [inline test/utils/runners.go]: GpuLimit

File: test/utils/runners.go (hunk 2 above)
Comment (losipiuk) [reply test/utils/runners.go]: It is defined. But it is defined in e2e/framework module. Which does not seem like a nice dependency of test/runners module. Hence the question.

I will of course drop the TODO before it is merged. I put it just for sake of review so we settle out the correct code shape.

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 4]
```diff
@@ -207,6 +207,76 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
...
+
+		installNvidiaDriversDaemonSet()
+
+		By("Schedule a single pod which requires GPU")
+		framework.ExpectNoError(scheduleGpuPod(f, "gpu-pod-rc"))
+
+		By("Enable autoscaler")
+		framework.ExpectNoError(enableAutoscaler(gpuPoolName, 0, 1))
+		defer disableAutoscaler(gpuPoolName, 0, 1)
+		Expect(len(getPoolNodes(f, gpuPoolName)), 1)
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: Expect(...).Should(Equal(...))

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 5]
```diff
@@ -207,6 +207,76 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
...
+
+		installNvidiaDriversDaemonSet()
+
+		By("Schedule a single pod which requires GPU")
+		framework.ExpectNoError(scheduleGpuPod(f, "gpu-pod-rc"))
+
+		By("Enable autoscaler")
+		framework.ExpectNoError(enableAutoscaler(gpuPoolName, 0, 2))
+		defer disableAutoscaler(gpuPoolName, 0, 2)
+		Expect(len(getPoolNodes(f, gpuPoolName)), 1)
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: Expect(...).Should(Equal(...))

File: test/e2e/autoscaling/cluster_size_autoscaling.go [hunk 6]
```diff
@@ -207,6 +207,76 @@ var _ = SIGDescribe("Cluster size autoscaling [Slow]", func() {
...
+		const gpuPoolName = "gpu-pool"
+		addGpuNodePool(gpuPoolName, "nvidia-tesla-k80", 1, 0)
+		defer deleteNodePool(gpuPoolName)
+
+		installNvidiaDriversDaemonSet()
+
+		By("Enable autoscaler")
+		framework.ExpectNoError(enableAutoscaler(gpuPoolName, 0, 1))
+		defer disableAutoscaler(gpuPoolName, 0, 1)
+		Expect(len(getPoolNodes(f, gpuPoolName)), 0)
```
Comment (aleksandra-malinowska) [inline test/e2e/autoscaling/cluster_size_autoscaling.go]: Expect(...).Should(Equal(...))

File: test/e2e/autoscaling/cluster_size_autoscaling.go (hunk 4 above)
Comment (losipiuk) [reply test/e2e/autoscaling/cluster_size_autoscaling.go]: Yeah - already fixed that.

File: test/utils/runners.go [hunk 7]
```diff
@@ -633,6 +634,10 @@ func (config *RCConfig) applyTo(template *v1.PodTemplateSpec) {
 	if config.MemRequest > 0 {
 		template.Spec.Containers[0].Resources.Requests[v1.ResourceMemory] = *resource.NewQuantity(config.MemRequest, resource.DecimalSI)
 	}
+	if config.GpuLimit > 0 {
+		template.Spec.Containers[0].Resources.Limits["nvidia.com/gpu"] = *resource.NewQuantity(config.GpuLimit, resource.DecimalSI)
+		// TODO[losipiuk] should I use framework.NVIDIAGPUResourceName from gpu_util.go as key?
```
Comment (aleksandra-malinowska) [inline test/utils/runners.go]: clean up todo

**Comment by losipiuk** [comment]
@shyamjvs could you take a look at small framework changes and approve this PR?

File: test/utils/runners.go (hunk 7 above)
Comment (shyamjvs) [reply test/utils/runners.go]: +1
